src/charm.py

```diff
@@ -431,7 +431,16 @@ def _on_set_password(self, event: ActionEvent) -> None:
 
     def _get_cluster_status(self, event: ActionEvent) -> None:
         """Get the cluster status without topology."""
-        event.set_results(self._mysql.get_cluster_status())
+        status = self._mysql.get_cluster_status()
+        if status:
+            event.set_results(status)
+        else:
+            event.set_results(
+                {
+                    "error": "Failed to get cluster status.  "
+                    "See logs for more information and traceback.",
+                }
+            )
```

Comment:
i would suggest standardizing the result structure for both success and error scenarios. 

upon success:
```
{
    "success": true,
    "status": status_dictionary
}
```

upon error:
```
{
    "success": false
}
```
this enables the caller to correctly and confidently condition on the `success` of the action

Reply:
I wouldn't sweat it too much if it's impossible to get the type of the field to be a boolean (at least we have a string with consistent values that we can rely on to condition upon)